opentelemetry-semantic-conventions/src/opentelemetry/semconv/v1_23_1/http_metrics.py

```diff
@@ -0,0 +1,109 @@
+
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from opentelemetry.metrics import (
+    Counter,
+    Histogram,
+    Meter,
+    UpDownCounter,
+    ObservableGauge,
+)
+from typing import Callable, Sequence
+
+class HttpMetrics:
+
+    """
+    Size of HTTP client request bodies
+    """
+    @staticmethod
+    def create_http_client_request_body_size(meter: Meter) -> Histogram:
```

Comment:
Might be a dumb question but why do we have a function with a Histogram here? I thought these were semantic conventions files.

Reply:
This is a demonstration of jinja template capabilities - you can generate metrics like that, otherwise it would make sense to generate a bunch of constants:

```python
HTTP_CLIENT_REQUEST_DURATION_METRIC_NAME = "...."
HTTP_CLIENT_REQUEST_DURATION_DESCRIPTION = "...."
HTTP_CLIENT_REQUEST_DURATION_UNIT = "s"
```

or `http_metrics.py`

```python
class HttpClientRequestDuration:
    METRIC_NAME = "...."
    DESCRIPTION = "...."
    UNIT = "s"
```

All of these options are possible, but they won't provide the instrument and caller will be responsible to use every constant in the right place

Reply:
Discussed at 3/14 SIG meeting, there is a tentative agreement on:

- generating metric names constants in all artifacts 

```python
HTTP_CLIENT_REQUEST_DURATION = "http.client.request.duration"
"""
Measures the duration of inbound HTTP requests
Instrument: histogram
Unit: s
"""
```

- generating metric factory functions in the experimental artifact only 
  - they should be top-level functions and not static methods
  - we're not sure about the final shape of this API and want to experiment a bit before putting it into stable arifact

I'll update the PR with this.